lib/rex/proto/smb/simpleclient.rb

```diff
@@ -32,8 +33,14 @@ class SimpleClient
   def initialize(socket, direct = false)
     self.socket = socket
     self.direct = direct
-    self.client = Rex::Proto::SMB::Client.new(socket)
-    self.shares = { }
+
+    self.client = RubySMB::Client.new(RubySMB::Dispatcher::Socket.new(self.socket, read_timeout: 60),
+                                      username: '',
+                                      password: '')#Rex::Proto::SMB::Client.new(socket)
+    self.client.class.module_eval { attr_accessor :evasion_opts}
```

Comment:
This looks a bit archaic, and like the namespace would be hit on every run. Or is the target lexical scope (client.class) dynamically composed for each run and this requiring the eval?

Reply:
This isn't right, I agree. My temporary reasoning for this line was, metasploit-framework uses the :evasion_opts attribute. The RubySMB client however is more general purpose, so it didn't seem to belong as an attribute. It would absolutely make sense to define this attribute outside of SimpleClient's #initialize and not dynamically compose on each run. The general approach taken here was to add the missing methods/attributes to the RubySMB client that Metasploit used with Rex::Proto::SMB::Client such that enumshare, enumusers, psexec would work with SMB2. It was mostly adding methods to RubySMB client to call the library's already existing file open/read/write/close operations (for writing to named pipes/dcerpc).